src/hathor.c

```diff
@@ -65,33 +65,29 @@ void derive_private_key(cx_ecfp_private_key_t *private_key,
                         uint8_t chain_code[static 32],
                         const uint32_t *bip32_path,
                         uint8_t bip32_path_len) {
-    uint8_t raw_private_key[32] = {0};
+    cx_err_t error = CX_OK;
+    uint8_t raw_privkey[65];
 
-    BEGIN_TRY {
-        TRY {
-            // derive the seed with 44'/280'/$(bip32_path)
-            os_perso_derive_node_bip32(CX_CURVE_256K1,
-                                       bip32_path,
-                                       bip32_path_len,
-                                       raw_private_key,
-                                       chain_code);
-            // new private_key from raw
-            cx_ecfp_init_private_key(CX_CURVE_256K1,
-                                     raw_private_key,
-                                     sizeof(raw_private_key),
-                                     private_key);
-        }
-        CATCH_OTHER(e) {
-            THROW(e);
-        }
-        FINALLY {
-            explicit_bzero(&raw_private_key, sizeof(raw_private_key));
-        }
+    CX_CHECK(os_derive_bip32_with_seed_no_throw(HDW_NORMAL,
```

Comment:
`CX_CHECK(a)` is a macro that jumps to `end` if a call returns something other than `CX_OK` (or 0) and sets `error` to the value.
This is how we achieve the same `finally` behavior of the `try..catch`.

OBS: Added security point, any errors now also clean the `private_key` buffer before throwing.